### PR TITLE
feat: added function get_subscriptions_by_entity with test

### PR DESCRIPTION
**Pull request by sbanoeon**
For #141

### CODE REVIEW

**Comment by tstorek** [comment]
@sbanoeon thank you for this contribution. I think this is related to #141 and #137. 
Please, check if this should not better become a filter function in utils. Since it actually copies the code from `get_subscriptions_list`. Hence, let's rather prepare the filter function that can be used for filtering matching subscriptions on multiple levels from entity matching down to complete logic matching :)

File: filip/utils/filter.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                                   url: str = None,
+                                   fiware_header: FiwareHeader = None,
+                                   subscriptions: List[Subscription] = None,
+                                   ) -> List[Subscription]:
+    """
+    Function that filters subscriptions based on the entity id or id pattern
+    and entity type or type pattern. The function can be used in two ways,
+    wither pass list of subscriptions to filter based on entity or directly pass
+    client information to filter subscriptions in a single request.
+    Args:
```
Comment (tstorek) [inline filip/utils/filter.py:21]: missing blankline

**Review by tstorek — CHANGES_REQUESTED** [review]
@sbanoeon looks quite good. please add the corresponding test

File: tests/utils/test_filter.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"""
```
[the remark this answers is not in the document]
Comment (sbanoeon) [reply tests/utils/test_filter.py:1]: You mean test_subscriptions.py, right?

[the remark this answers is not in the document]
Comment (tstorek) [reply tests/utils/test_filter.py:1]: oh sorry. I meant both the test file and the module in filip.utils.

File: tests/utils/test_filter.py [hunk 3]
```diff
@@ -2,9 +2,86 @@
...
+from filip.clients.ngsi_v2 import ContextBrokerClient
+from filip.models.base import FiwareHeader
+from filip.models.ngsi_v2.base import EntityPattern
+from filip.models.ngsi_v2.subscriptions import Subscription
+from filip.utils import filter
+from filip.utils.cleanup import clear_all
+from tests.config import settings
 
 
 class TestFilterFunctions(unittest.TestCase):
```
Comment (tstorek) [inline tests/utils/test_filter.py:16]: please have a look at utils.iot there you will find the other filter I mentioned

**Review by tstorek — CHANGES_REQUESTED** [review]
@sbanoeon looks quite good. please. make the last changes :)

File: tests/utils/test_filter.py (hunk 3 above)
Comment (sbanoeon) [reply tests/utils/test_filter.py:16]: Then I think the best place for this testcase is in `tests.clients.test_ngsi_v2_cb.py`. What do you think? Or it should be separate in `tests.utils`?

**Pull request merged**